### PR TITLE
feat: decouple functions in structs from solana env + add tests

### PR DESCRIPTION
**Pull request by keyvankhademi**

### CODE REVIEW

File: staking/programs/staking/src/state/positions.rs [hunk 1]
```diff
@@ -33,7 +33,8 @@ impl PositionData {
     pub const LEN: usize = 8 + 32 + MAX_POSITIONS * POSITION_BUFFER_SIZE;
 }
 
-#[cfg(test)]
+// TODO: fix this
+// #[cfg(test)]
```
Comment (keyvankhademi) [inline staking/programs/staking/src/state/positions.rs:37]: compiler gives and error

Comment (guibescos) [reply staking/programs/staking/src/state/positions.rs:37]: tbh don't mind getting rid of this. Not a big fan of these conditional implementations if we are in test mode anyway

File: staking/programs/integrity-pool/src/lib.rs [hunk 2]
```diff
@@ -226,18 +225,19 @@ pub mod integrity_pool {
         let token_program = &ctx.accounts.token_program;
         let publisher = &ctx.accounts.publisher;
 
-        let reward_amount_frac: frac64 = pool_data.calculate_reward(
+        // reward amount in PYTH with decimals
+        let reward_amount: frac64 = pool_data.calculate_reward(
```
Comment (guibescos) [inline staking/programs/integrity-pool/src/lib.rs:229]: so frac64 is defined as a number with 6 decimals?

**Review by guibescos — APPROVED** [review]
lgtm

File: staking/programs/integrity-pool/src/lib.rs (hunk 2 above)
Comment (keyvankhademi) [reply staking/programs/integrity-pool/src/lib.rs:229]: yes. we have to treat it like float.

**Pull request merged**